github/util.py

```diff
@@ -95,6 +95,19 @@ def create_or_update_file(
 
         return response["commit"].sha
 
+    def retrieve_text_file_contents(
```

Comment:
use this new helper method in 'create_or_update_file', as otherwise, the introduction of this method would duplicate code.

while doing so, please consider:

1. renaming GithubHelper to something more appropriate (like GithubRepositoryHelper)
2. passing the repository branch via the c'tor (and rm it from all offered methods)
3. consider accepting a github_cfg object in the c'tor rather than a github_api object (need to have a look at existing code to judge whether this is a good idea)